src/services/__tests__/QueueService.int.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { joinQueue } from "../QueueService";
+import { leaveQueue } from "../QueueService";
+import { BallChaserQueueBuilder, LeaderboardBuilder } from "../../../.jest/Builder";
+import { DateTime } from "luxon";
+import getEnvVariable from "../../utils/getEnvVariable";
+import { PrismaClient } from "@prisma/client";
+import LeaderboardRepository from "../../repositories/LeaderboardRepository";
+import { mocked } from "ts-jest/utils";
+import { join } from "path/posix";
+
+jest.mock("../../utils/getEnvVariable");
+jest.mock("../../repositories/LeaderboardRepository");
+
+let prisma: PrismaClient;
+let seasonSemester: string;
+let seasonYear: string;
+
+beforeEach(async () => {
+  jest.clearAllMocks();
+});
+
+beforeAll(async () => {
+  seasonSemester = getEnvVariable("season_semester");
+  seasonYear = getEnvVariable("season_year");
+
+  prisma = new PrismaClient();
+  await prisma.$connect();
+  await prisma.leaderboard.deleteMany();
+  await prisma.activeMatch.deleteMany();
+  await prisma.queue.deleteMany();
+  await prisma.ballChaser.deleteMany();
+});
+
+afterEach(async () => {
+  await prisma.leaderboard.deleteMany();
+  await prisma.activeMatch.deleteMany();
+  await prisma.queue.deleteMany();
+  await prisma.ballChaser.deleteMany();
+});
+
+afterAll(async () => {
+  await prisma.$disconnect();
+});
+
+const mockPlayer1 = BallChaserQueueBuilder.single({});
+
+describe("QueueService tests", () => {
+  describe("Joining queue", () => {
+    describe("Not already in queue", () => {
+      it("joins a player to the queue if they are not already in the queue", async () => {
+        const result = await joinQueue(mockPlayer1.id, mockPlayer1.name);
+
+        expect(result).toHaveLength(1);
+      });
+
+      it("Not on leaderboard | joins queue with 100 MMR and queue time 1 hour from now", async () => {
+        const result = await joinQueue(mockPlayer1.id, mockPlayer1.name);
+        let mmr = result[0].mmr;
```

Comment:
These variables aren't being reassigned so you should use const instead.

Reply:
Resolved.